pkg/errcode/errcode.go

```diff
@@ -373,6 +374,13 @@ func (e ErrCode) String() string {
 	return fmt.Sprintf("E%d", e)
 }
 
+// GetErrCodeWithMetric increments the ErrcodeCounter metric for the given error code
+// Returns the error code as a string
+func GetErrCodeWithMetric(e ErrCode) string {
```

Comment:
Let's add a small unit test for this (no need to verify that metrics work).

Reply:
Added!